qip-0011.md

```diff
@@ -0,0 +1,209 @@
+
+The reward fuction for Quai will take the form of:
+
+$$ Quai(k_{quai}(\bar{R}\_{hash}),diff) = 2^{-k_{quai}(\bar{R}_{hash})} * log_2(diff) $$
+
+The astutue reader will notice that a simple model has been introduced, namely raising $k_{quai}$
+to the power of 2. This will allow a simple controller to maintain stability
+with fixed controller gains over orders of magnitude changes in difficulty.
+
+### Projected  Efficiency ###
+
+The projected GPU efficiency is introduced to help Qi be more strictly reflective of the electricity
+cost component in mining.  This will help Qi maintain a more stable purchasing power compared
+to having a reward function that does not adjust with GPU efficiency. Again the primary goal
+being to have a low enough volotility in the change in purchasing power to allow goods and
+services to be easily denominatable in Qi. Thus, Qi could be used as a unit of account.
+
+The projected GPU efficiency curve should be fitted to relect the median prediction from:
+
+Marius Hobbhahn and Tamay Besiroglu (2022), ["Predicting GPU Performance"](https://epochai.org/blog/predicting-gpu-performance)
+
+The curve proposed is as follows:
+
+![projected-gpu-performance](qip-0011/predicting-gpu-performance.png "Projected GPU performance")
+> Image 1: The projected performance of FET based GPUs
+
+Extracted Data 
+```
+X | Y
+2002.7,10.75
+2014.1,12.04
+2024.75,13.36
+2028.9,13.87
+2030.8,14.10
+2032.6,14.23
+2036.7,14.31
+2051.4,14.31
+```
+The fit derived from the expected performance data is as follows:
+
+$$
+f(x) = \begin{cases} 
+-229.16 + 0.119779 x & \text{for } x < 2032 \\
+14.257 & \text{for } x \geq 2032 \\
+\end{cases}
+$$
+
+Based on the launch date of quai this function will need to be adjusted to account
+for both the start date as well as the conversion between block number and year.
+
+## Controller Design
+
+Both the exchange rate and block reward controller will take the form of a simple
+proportional integral controller.
+
+### Exchange Rate Controller ###
+
+The exchange rate controller input will be the simple moving average of the 
+aggregate hash equivalent demand, $\bar{R}\_{hash}$ for Qi, $D_{qi}$, and Quai, $D_{quai}$ over a period of 100 prime blocks. The setpoint
+of the controller will be to maintain equivalent aggregate demand for Qi and Quai. The hash equivalent value
+needs to be updated with each step. The $\bar{R}_{hash}$ will be computed as follows:
+
+$$ \bar{R}\_{hash} = \sum_{l=n-m}^{n} \frac{D_{quai,l} - D_{qi,l}}{D_{quai,l} + D_{qi,l}} $$
+
+where $n$ is the block number, $m= 100$ is the depth over which the average is taken. This will create an output that ranges from -1 to 1. This will be the input on which the controller operates. The controller for computing $k_{quai,exchange}(\bar{R}_{hash})$ will take the following form:
+
+$$ k_{quai,exchange}(\bar{R}\_{hash}) = k_{quai,exchange}(\bar{R}\_{hash}) + k_{quai}(\bar{R}\_{hash}) * (P * \bar{R}\_{hash} + I * \sum_{l=n-m}^{n} \bar{R}\_{hash}) $$
```

Comment:
This equation is infinitely recursive:

$$ k(\bar{R}\_{hash}) = k(\bar{R}\_{hash}) + f(\bar{R}\_{hash}) $$

Reply:
need to add n and n-1

Reply:
I think the index is actually on $\bar{R}\_{hash}$, i.e. $\bar{R}\_{hash}[n]$